File: client-flutter/lib/services/auth_service.dart
import '../models/auth_token.dart';
import 'api_client.dart';

class AuthService {
  Future<AuthToken> login(String email, String password) async {
    final data = await apiClient.post('api/client/token', body: {
      'email': email,
      'passcode': password,
    });
    if (data == null || data is! Map<String, dynamic>) {
      throw ApiException(401, 'Login fehlgeschlagen');
    }
    return AuthToken.fromJson(data);
  }
}

File: client-ios/SihlClient/Auth/AuthService.swift
import Foundation

/// Pendant zu Flutter `auth_service.dart`.
/// Kapselt den Login-Request gegen `POST api/client/token`.
struct AuthService {
    private let api = APIClient.shared

    func login(email: String, password: String) async throws -> AuthToken {
        let data = try await api.post("api/client/token", body: [
            "email": email,
            "passcode": password,
        ])
        do {
            return try JSONDecoder().decode(AuthToken.self, from: data)
        } catch {
            throw APIError(statusCode: 401, message: "Login fehlgeschlagen")
        }
    }
}
